environments/prod/main.tf

```diff
@@ -84,11 +95,32 @@ module "container_orchestrator" {
   }
 
   network = {
-    # checkov:skip=CKV_AWS_88: "EC2 instance should not have public IP."
-    # For cost saving purpose we place instances in public subnet and assign public IP.
-    # Disabling public ips would require NAT or PrivateLink to communicate with ECS
-    assign_instance_public_ips = true
-    subnets                    = module.network.public_subnets
+    assign_instance_public_ips = false
+    subnets                    = module.network.private_subnets
     security_groups            = [module.network.default_security_group]
   }
 }
+
+module "app_api" {
+  source = "../../modules/app_api"
+  name   = "${var.stack_name}-api"
+
+  cluster_id = module.container_orchestrator.cluster_id
+  vpc_id     = module.network.vpc_id
+}
+
+module "load_balancer" {
+  source = "../../modules/load_balancer"
+  name   = "${var.stack_name}-default-alb"
+
+  network = {
+    vpc_id          = module.network.vpc_id
+    private_subnets = module.network.private_subnets
+    public_subnets  = module.network.public_subnets
+    security_groups = [module.network.default_security_group] #TODO: verify if we can allow less
+  }
+  default_target_group_arn = module.app_api.target_group_arn
```

Comment:
Can we decouple it using [lb_target_group_attachment](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_target_group_attachment)?

Reply:
Would be great to have a chat with you about that - pros / cons.

Reply:
What advantages do you see with the proposed approach?

The already implemented solution is using default action for ALB which could not be defined in the proposed way 😢 

If we will want to add more targets, then sure - this is the best way.

Reply:
During the discussion, we agreed that this will not help.